File: src/main/java/com/example/RESTAPIarticle/service/AuthorServiceImpl.java
package com.example.RESTAPIarticle.service;

import com.example.RESTAPIarticle.dao.AuthorDAO;
import com.example.RESTAPIarticle.entity.Author;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class AuthorServiceImpl implements AuthorService{
    private final AuthorDAO authorDAO;
    @Override
    public Author findById(int theId) {
        return authorDAO.findById(theId).orElse(null);
    }

    @Override
    public void save(Author theAuthor) {
        theAuthor.setId(0);
        authorDAO.save(theAuthor);
    }

    @Override
    public Author findByFirstNameAndLastName(String firstName, String lastName) {
        return authorDAO.findByFirstNameAndLastName(firstName,lastName);
    }
}

File: src/main/java/com/example/RESTAPIarticle/service/AuthorServiceImpl.kt
package com.example.RESTAPIarticle.service

import com.example.RESTAPIarticle.entity.Author
import com.example.RESTAPIarticle.repository.AuthorRepository
import lombok.RequiredArgsConstructor
import org.springframework.stereotype.Service

@Service
@RequiredArgsConstructor
class AuthorServiceImpl(private val authorRepository: AuthorRepository) : AuthorService {

    override fun findById(theId: Int): Author? {
        return authorRepository.findById(theId).orElse(null)
    }

    override fun save(theAuthor: Author) {
        theAuthor.id = 0
        authorRepository.save(theAuthor)
    }

    override fun findByFirstNameAndLastName(firstName: String, lastName: String): Author? {
        return authorRepository.findByFirstNameAndLastName(firstName, lastName)
    }
}